packages/flutter/lib/src/widgets/sliver_resizing_header.dart

```diff
@@ -0,0 +1,232 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'basic.dart';
+import 'focus_scope.dart';
+import 'framework.dart';
+import 'slotted_render_object_widget.dart';
+
+/// A sliver that is pinned to the start of its [CustomScrollView] and
+/// reacts to scrolling by resizing between the intrinsic sizes of its
+/// min and max extent prototypes.
+///
+/// The minimum and maximum sizes of this sliver are defined by [minExtentPrototype]
+/// and [maxExtentPrototype], a pair of widgets that are laid out once. You can
+/// use [SizedBox] widgets to define the size limits.
+///
+/// If the [minExtentPrototype] is null, then the default minimum extent is 0. If
+/// [maxExtentPrototype] is null then the default maximum extent is based on the child's
+/// intrisic size.
+///
+/// This sliver is preferable to the general purpose [SliverPersistentHeader]
+/// for its relatively narrow use case because there's no need to create a
+/// [SliverPersistentHeaderDelegate] or to predict the header's minimum or
+/// maximum size.
+///
+/// {@tool dartpad}
+/// This sample shows how this sliver's two extent prototype properties can be used to
+/// create a resizing header whose minimum and maximum sizes match small and large
+/// configurations of the same header widget.
+///
+/// ** See code in examples/api/lib/widgets/sliver/sliver_resizing_header.0.dart **
+/// {@end-tool}
+// TODO(hansmuller): add See also links to PersistentHeaderSliver, SliverFloatingHeader, etc
+class SliverResizingHeader extends StatelessWidget {
```

Comment:
Since this is a pinned sliver, and there is another PR of this set that is pinned and includes `Pinned`, I still feel like it should have `Pinned` in the title, but I'll defer to your judgement.

Reply:
Just to support the idea a bit further, folks searching for Pinned in the API docs will find the other one, and not this option.

Reply:
Header also means "appears at the top" and I'd really prefer to avoid long compound names. In this case "resizing" emphasizes what's  special about this header, as pinned does for PinnedHeaderSliver. I'll update the see also links in both classes to make sure that if someone finds the wrong header it will be a short trip to find the correct one.